bundles/framework/publisher2/instance.js

```diff
@@ -94,14 +93,17 @@ Oskari.clazz.define('Oskari.mapframework.bundle.publisher2.PublisherBundleInstan
          * @method afterStart
          */
         afterStart: function () {
+            var me = this;
             var sandbox = this.getSandbox();
             var loc = this.getLocalization();
 
             this.__service = Oskari.clazz.create('Oskari.mapframework.bundle.publisher2.PublisherService', sandbox);
             // create and register request handler
             var reqHandler = Oskari.clazz.create(
-                    'Oskari.mapframework.bundle.publisher.request.PublishMapEditorRequestHandler',
-                    this);
+                'Oskari.mapframework.bundle.publisher.request.PublishMapEditorRequestHandler',
+                function(data) {
+                    me.setPublishMode(true, data);
```

Comment:
We could just pass bound `setPublishMode` to the PublishMapEditorRequestHandler constructor:
`this.setPublishMode.bind(this, true)`
but maybe your version is more readable.